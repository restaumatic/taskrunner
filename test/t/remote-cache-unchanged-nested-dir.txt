# no toplevel
# s3

# Check how caching works when we're not running directly in root dir.
# We should still be able to cache stuff outside cwd (but inside root dir).

export TASKRUNNER_SAVE_REMOTE_CACHE=1

mkdir -p a

(
  cd a
  git init -q
  mkdir nested
  cd nested
  echo foo > input.txt
  git add input.txt
  git commit -qm "Add input.txt"
)

cp -r a b

go() {
  # Note: we want separate workdir for separate repos
  export TASKRUNNER_STATE_DIRECTORY="$(pwd)" TASKRUNNER_ROOT_DIRECTORY="$(pwd)"
  (
    cd nested
    taskrunner -n mytask bash -e -c '
      snapshot input.txt --outputs output.txt ../output2.txt
      echo "Expensive computation"
      rev input.txt > output.txt
      cp output.txt ../output2.txt
    '
  )
}

(
  cd a
  go
  echo "Output in a is: $(cat nested/output.txt)"
  echo "Output2 in a is: $(cat output2.txt)"
)

(
  cd b
  go ../out_b # note: should not compute again
  echo "Output in b is: $(cat nested/output.txt)"
  echo "Output2 in b is: $(cat output2.txt)"
)
