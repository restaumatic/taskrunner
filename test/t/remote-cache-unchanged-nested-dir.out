-- output:
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation
[mytask] info | success
Output in a is: oof
Output2 in a is: oof
[mytask] info | Restored from remote cache
Output in b is: oof
Output2 in b is: oof
